packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -143,8 +144,51 @@ namespace Amesos2 {
     // Redistribute the input (multi)vector.
     redist_mv.doExport (*mv_, *exporter_, Tpetra::REPLACE);
 
+    // Do this if GIDs not contiguous...
+    // getLocalView, copy into Teuchos::ArrayView av
+    // Need safety about the sizes...
+    typedef typename multivec_t::dual_view_type dual_view_type;
+    typedef typename dual_view_type::host_mirror_space host_execution_space;
+    redist_mv.template sync < host_execution_space > ();
+    
+    //auto myRank = distMap->getComm()->getRank();
+    auto contig_local_view_2d = redist_mv.template getLocalView<host_execution_space>();
```

Comment:
You should only sync to host if the solver itself demands it, though I suppose that is every solver that Amesos2 now uses ;-)